### ISSUE TITLE
How should I be running the Enhanced BERT Baseline model?

### ISSUE THREAD

**Issue by seanswyi** [wzhouad/ATLOP#17]
Hi. I recently tried to run the Enhanced BERT Baseline model (i.e., without adaptive threshold loss and local contextualized pooling) and just wanted to confirm if I'm doing it right.

Basically, in `model.py` lines 86-111 (i.e., the `forward` method) I modified the code so that I don't use `rs` and changed `self.head_extractor` and `self.tail_extractor` to have `in_features` and `out_features` accordingly. I did this because I'm assuming that within the `get_hrt` method, `rs` is what LOP is since we're using attention there. Modifying the extractors also implies that I'm not concatenating `hs` and `ts` with `rs`.

After that I changed `loss_fnt` to be a simple `nn.BCEWithLogitsLoss` rather than `ATLoss`. That means I also changed the `get_label` method within `ATLoss` to be a function so that I'm not depending on the class.

Am I doing this right? Or is there another way that I should be implementing it?

The reason why I'm suspicious as to whether I implemented this correctly or not is because I'm currently running the code on the TACRED dataset rather than the DocRED dataset, and while ATLOP itself shows satisfactory performance the performance of the Enhanced BERT Baseline is much lower.

Thanks.

**Comment by wzhouad** [comment]
Your modification looks correct to me. I don't think the enhanced BERT baseline is a good fit for TACRED for the following reasons:

(1) NER types of entities are strong clues for TACRED. It's better to add NER types to the input sequence instead of using entity markers. Check this [paper](https://arxiv.org/abs/2102.01373) for the comparison.

(2) The enhanced BERT baseline uses a bilinear classifier. It's effective for document-level RE because the task is to predict the pairwise relationships of entities and the bilinear learns the factorization, while for TACRED there's only one entity pair. I observed that the bilinear classifier performs worse than the linear classifier on TACRED.

(3) BCE is applied on DocRED as one entity pair may have multiple relationships in the document. While for TACRED one sentence only has one label. I assume softmax cross-entropy is a better choice.

**Comment by seanswyi** [comment]
Thanks for the tips! I'll take a closer look into your suggestions.
